pkg/specgen/generate/namespaces.go

```diff
@@ -2,293 +2,421 @@ package generate
 
 import (
 	"os"
+	"strings"
 
 	"github.com/containers/common/pkg/capabilities"
 	"github.com/containers/libpod/libpod"
+	"github.com/containers/libpod/libpod/define"
 	"github.com/containers/libpod/libpod/image"
+	"github.com/containers/libpod/pkg/cgroups"
+	"github.com/containers/libpod/pkg/rootless"
 	"github.com/containers/libpod/pkg/specgen"
-	"github.com/cri-o/ocicni/pkg/ocicni"
 	spec "github.com/opencontainers/runtime-spec/specs-go"
 	"github.com/opencontainers/runtime-tools/generate"
 	"github.com/pkg/errors"
 	"github.com/sirupsen/logrus"
 )
 
-func GenerateNamespaceContainerOpts(s *specgen.SpecGenerator, rt *libpod.Runtime) ([]libpod.CtrCreateOption, error) {
-	var portBindings []ocicni.PortMapping
-	options := make([]libpod.CtrCreateOption, 0)
-
-	// Cgroups
-	switch {
-	case s.CgroupNS.IsPrivate():
-		ns := s.CgroupNS.Value
-		if _, err := os.Stat(ns); err != nil {
-			return nil, err
+// Get the default namespace mode for any given namespace type.
+// TODO: Do we want to wire this into containers.conf? I lean towards no, that's
```

Comment:
There are calls to do this in containers.conf now.

Reply:
Containers.conf doesn't really parse directly into namespaces at this level, which is why I'm hesitant.

Reply:
Users want to to be able to disable namespaces.  Something like this, for running podman inside of a container.
```
# Default way to to create a cgroup namespace for the container
# Options are:
# `private` Create private Cgroup Namespace for the container.
# `host`    Share host Cgroup Namespace with the container.
#
cgroupns = "host"
```

Reply:
Which will happen on the client side and will hand off to us a properly-formatted namespace. I don't want this code to have to parse strings.